nautobot/users/tests/test_api.py

```diff
@@ -73,6 +73,173 @@ def setUpTestData(cls):
+
+        tokens = [
+            # We already start with one Token, created by the test class
+            Token.objects.create(user=self.user),
+            Token.objects.create(user=self.user),
+        ]
+
+        self.tokens = tokens + [self.token]
+
+        self.basic_auth_user_password = "abc123"
+        self.basic_auth_user_granted = User.objects.create_user(
+            username="basicusergranted", password=self.basic_auth_user_password
+        )
+
+        obj_perm = ObjectPermission(name="Test permission", actions=["add", "change", "view", "delete"])
+        obj_perm.save()
+        obj_perm.users.add(self.basic_auth_user_granted)
+        obj_perm.object_types.add(ContentType.objects.get_for_model(self.model))
+
+        self.basic_auth_user_permissionless = User.objects.create_user(
+            username="basicuserpermissionless", password=self.basic_auth_user_password
+        )
+
+        self.create_data = [
+            {
+                "description": "token1",
+            },
+            {
+                "description": "token2",
+            },
+            {
+                "description": "token3",
+            },
+        ]
+
+    def test_create_token_basic_authentication(self):
+        """
+        Test the provisioning of a new REST API token given a valid username and password.
+        """
+        self.client.login(username=self.basic_auth_user_granted.username, password=self.basic_auth_user_password)
+        response = self.client.post(self._get_list_url())
+        self.client.logout()
+
+        self.assertEqual(response.status_code, 201)
+        self.assertIn("key", response.data)
+        self.assertEqual(len(response.data["key"]), 40)
+        token = Token.objects.get(user=self.basic_auth_user_granted)
+        self.assertEqual(token.key, response.data["key"])
+
+    def test_create_token_basic_authentication_permissionless_user(self):
+        """
+        Test the behavior of the token create view when a user cannot create tokens
+        """
+        self.client.login(username=self.basic_auth_user_permissionless.username, password=self.basic_auth_user_password)
+        response = self.client.post(self._get_list_url())
+        self.client.logout()
+
+        self.assertEqual(response.status_code, 403)
+
+    def test_create_token_basic_authentication_invalid_password(self):
+        """
+        Test the behavior of the token create view when an invalid password is supplied
+        """
+        self.client.login(username=self.basic_auth_user_granted.username, password="hunter2")
+        response = self.client.post(self._get_list_url())
+        self.client.logout()
+
+        self.assertEqual(response.status_code, 403)
+
+    def test_create_token_basic_authentication_invalid_user(self):
+        """
+        Test the behavior of the token create view when the user supplied is not a valid user
+        """
+        self.client.login(username="iamnotreal", password="P1n0cc#10")
```

Comment:
Likewise here - is this redundant with the previous test case, since both are really just testing what happens when you hit the endpoint without logging in successfully first?